### DISCUSSION TITLE
Coding Challenge part 4 - added challenge?

### DISCUSSION THREAD

**Discussion by BioNCphilic** [earthlab-education/Earth-Analytics-AY24#377] [Q&A] [answered]
I'm trying to eliminate the daily temperatures before 1900 (1/1/1900) from my graph. Here's the code I'm using, but there are errors. I believe line 4 is the issue. Any suggestions for improvement? Thanks in advance!

1 # Plot the data using .plot
2 # Alter date range on plot
3 
4 climate_df = pd.DataFrame({
 5   'date': pd.date_range('1900-01-01', '2023-09-30', freq='D'),
6 })
7
8 climate_df.plot(
 9   ax = df.plot(x='date', y='value'),
 10   y='TOBS',
 11   title = 'Daily Temperature in Boulder CO',
 12   xlabel = 'Date',
 13   ylabel = 'degrees Fahrenheit')
14 plt.show()
15

**Comment by lauren-pease** [comment]
@BioNCphilic You may want to identify and supply the number of periods to [pd.date_range()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html). It's possible there is missing daily data, and so the method may not produce the intervals you're expecting. An easier way though may be to access the data using a condition with [pandas .loc[]](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html). Something like: df.loc[df['DATE'] >= '1900-01-01'].

**Comment by eculler** [comment, marked as answer]
@BioNCphilic -- so, it looks like you're generating a list of dates, rather than selecting data from your DataFrame. You want to be really careful with this approach, because you could easily end up mislabeling data.

Instead, I'd recommend using the .loc attribute to select dates -- for example climate_df.loc['1900': '2024'] should select the  dates between 1900 and 2024.
